config/routes.rb

```diff
@@ -4,6 +4,23 @@
   mount_devise_token_auth_for 'Valet', at: 'valets'
   as :valet do
     # Define routes for Valet within this block.
+
+
   end
   root 'static_pages#index'
+
+    # valet change profile info
+    patch '/api/v1/valets/:id' => 'valets#update'
```

Comment:
how about:

```ruby
namespace '/api/v1' do
  resources :valets, only: [:update, :show, :index] do
    collection do
      get 'available'
    end
  end
```
http://guides.rubyonrails.org/routing.html#controller-namespaces-and-routing